domains/client/block-preprocessor/src/stateless_runtime.rs

```diff
@@ -226,4 +226,19 @@ where
             block_hash,
         )
     }
+
+    pub fn check_transaction_validity_and_do_pre_dispatch(
```

Comment:
With a closer look, I think both `check_transaction_validity` and `check_transaction_validity_and_do_pre_dispatch` are not stateless runtime API, since they rely on the `nonce` and `balance` state for the nonce check and tx fee check. cc @vedhavyas

Reply:
That is correct, this check expects the storage proof for the keys read to be injected into StatelessApi before call this

Reply:
I see, it would be great if we could make it explicit or at least add some comment though.